nservicebus/nhibernate/custom-mappings.md

```diff
@@ -0,0 +1,126 @@
+
+## Use Fluent NHibernate
+
+Fluent NHibernate gives you a type-safe mapping approach where the mapping is specified as c# code and not as xml but still seperate from the classes. The benefit is that you get compile time feedback when a mapping is not valid any more when you have breaking changes in your mapping but also that your classes and mappings are not part of the same .net type.
+
+To use it with NServiceBus:
+
+1. Install `FluentNHibernate` package via Nuget.
+2. Create a custom NHibernate configuration
+a. via FluentNHibernate
+b. or by creating a new Configuration instance and pass it to FluentNHibernate
+3. Pass it to the NServiceBus NHibernate configuration.
+
+
+Example of a possible implementation:
+
+<!-- import NHibernateInitWithFluentNHibernate -->
+
+References:
+
+* [Fluent NHibernate - Website](http://www.fluentnhibernate.org)
+* [Fluent NHibernate - Nuget package](http://www.nuget.org/packages/FluentNHibernate/)
+* [Fluent NHibernate - GitHub Repository](https://github.com/jagregory/fluent-nhibernate)
+* [Fluent NHibernate - Getting Started](https://github.com/jagregory/fluent-nhibernate/wiki/Getting-started)
+
+
+## Use NHibernate.Mapping.Attributes
+
+With NHibernate.Mapping.Attributes you can decorate your saga data classes. This keeps your classes, mapping and schema data very close. Your saga types have a dependency on the NHibernate.Mapping.Attributes assembly.
+
+How NHibernate.Mapping.Attributes works is that it needs to know which types it needs to scan, then it create a regular NHibernate mapping configuration that gets passed to your NHibernate configuration.
+
+1. Add the nuget package `NHibernate.Mapping.Attributes`
+2. Create a custom NHibernate configuration object.
+3. Initialize the attribute mapping (see sample below).
+4. Pass it to the NServiceBus NHibernate configuration.
+
+
+Initialize the NHibernate attribute based mappings:
+
+<!-- NHibernateInitWithNHibernateMappingAttributes -->
+
+Take a look at the NHibernate mapping attributes documentation for an example.
+
+
+References:
+* [NHibernate.Mapping.Attributes - Documentation](http://nhibernate.info/doc/nhibernate-reference/mapping-attributes.html)
+* [NHibernate.Mapping.Attributes - GitHub repository](https://github.com/nhibernate/NHibernate.Mapping.Attributes)
+* [NHibernate.Mapping.Attributes - NuGet package](http://www.nuget.org/packages/NHibernate.Mapping.Attributes/)
+
+
+## Use the Loquacious API
+
+The Loquacious api is NHibernate its native fluent mapping ability. You declare your mapping in code just like FluentNHibernate but using a different syntax. It can help you create a type-safe configuration but you can also create your custom mappings. The benefit is that this api is already available via the NHibernate package so requires no additional downloads.
+
+Creating mappings with this API is very similar to its native xml mapping. Meaning that for each entity a mappings needs to be created.
```

Comment:
Creating mappings with the Loquacious API is similar to the native `xml` mappings: for each entity a mapping needs to be created.